In [15]:
import pandas as pd
import numpy as np
import seaborn as sns
from IPython.display import IFrame

In [16]:
IFrame('https://upload.wikimedia.org/wikipedia/en/1/18/Shark_Tank_%28ABC_Network_reality_series%29_logo.svg', width=350, height=270)

In [83]:
df = pd.read_csv(r"C:\Users\user\Desktop\Elevate Labs Internship\Projects\Project_1\ShartankIndiaAllPitches.csv")

In [11]:
df.head(5)

,Episode Number,Pitch Number,Brand,Idea,Investment Amount (In Lakhs INR),Debt (In lakhs INR),Equity,Anupam,Ashneer,Namita,Aman,Peyush,Vineeta,Ghazal,Season
0,1,1,BluePine Industries,Frozen Momos,75,0,18%,N,Y,N,Y,N,Y,N,1
1,1,2,Booz scooters,Renting e-bike for mobility in private spaces,40,0,50%,N,Y,N,N,N,Y,N,1
2,1,3,Heart up my Sleeves,Detachable Sleeves,25,0,30%,Y,N,N,N,N,Y,N,1
3,2,4,Tagz Foods,Healthy Potato Chips,70,0,2.75%,N,Y,N,N,N,N,N,1
4,2,5,Head and Heart,Brain Development Course,0,0,0,N,N,N,N,N,N,N,1


In [12]:
print(f"Rows {df.shape[0]}, columns {df.shape[1]}.")

Rows 117, columns 15.


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 15 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   Episode Number                     117 non-null    int64 
 1   Pitch Number                       117 non-null    int64 
 2   Brand                              117 non-null    object
 3   Idea                               117 non-null    object
 4   Investment Amount (In Lakhs INR)   117 non-null    int64 
 5   Debt (In lakhs INR)                117 non-null    int64 
 6   Equity                             117 non-null    object
 7   Anupam                             117 non-null    object
 8   Ashneer                            117 non-null    object
 9   Namita                             117 non-null    object
 10  Aman                               117 non-null    object
 11  Peyush                             117 non-null    object
 12  Vineeta 

In [14]:
df.isnull().sum()

Episode Number                       0
Pitch Number                         0
Brand                                0
Idea                                 0
Investment Amount (In Lakhs INR)     0
Debt (In lakhs INR)                  0
Equity                               0
Anupam                               0
Ashneer                              0
Namita                               0
Aman                                 0
Peyush                               0
Vineeta                              0
Ghazal                               0
Season                               0
dtype: int64

# There are 117 rows and 15 columns in the dataset.
# There are no null values in the dataset.
# The Datatype of `Equity` is object, that will be changed into float. 
# Also it has a extra space in `Investment Amount (In Lakhs INR)` column, that can show error while analyzing.

In [84]:
print(df.columns.tolist())

['Episode Number', 'Pitch Number', 'Brand', 'Idea', 'Investment Amount (In Lakhs INR) ', 'Debt (In lakhs INR)', 'Equity', 'Anupam', 'Ashneer', 'Namita', 'Aman', 'Peyush', 'Vineeta', 'Ghazal', 'Season']


In [88]:
df.rename(columns = {'Investment Amount (In Lakhs INR) ' : 'Investment Amount (In Lakhs INR)'}, inplace = True)
df

,Episode Number,Pitch Number,Brand,Idea,Investment Amount (In Lakhs INR),Debt (In lakhs INR),Equity,Anupam,Ashneer,Namita,Aman,Peyush,Vineeta,Ghazal,Season
0,1,1,BluePine Industries,Frozen Momos,75,0,18%,N,Y,N,Y,N,Y,N,1
1,1,2,Booz scooters,Renting e-bike for mobility in private spaces,40,0,50%,N,Y,N,N,N,Y,N,1
2,1,3,Heart up my Sleeves,Detachable Sleeves,25,0,30%,Y,N,N,N,N,Y,N,1
3,2,4,Tagz Foods,Healthy Potato Chips,70,0,2.75%,N,Y,N,N,N,N,N,1
4,2,5,Head and Heart,Brain Development Course,0,0,0,N,N,N,N,N,N,N,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
112,34,113,Green Protein,Plant-Based Protein,0,0,0,N,N,N,N,N,N,N,1
113,34,114,On2Cook,Fastest Cooking Device,0,0,0,N,N,N,N,N,N,N,1
114,35,115,Jain Shikanji,Lemonade,40,0,30.00%,Y,Y,N,Y,N,Y,N,1
115,35,116,Woloo,Washroom Finder,0,0,0,N,N,N,N,N,N,N,1


# Cleaned extra space in `Investment Amount (In Lakhs INR)` column

In [33]:
df['Equity'] = df['Equity'].str.replace('%','')
df['Equity'] = df['Equity'].astype('float64')

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 15 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Episode Number                     117 non-null    int64  
 1   Pitch Number                       117 non-null    int64  
 2   Brand                              117 non-null    object 
 3   Idea                               117 non-null    object 
 4   Investment Amount (In Lakhs INR)   117 non-null    int64  
 5   Debt (In lakhs INR)                117 non-null    int64  
 6   Equity                             117 non-null    float64
 7   Anupam                             117 non-null    object 
 8   Ashneer                            117 non-null    object 
 9   Namita                             117 non-null    object 
 10  Aman                               117 non-null    object 
 11  Peyush                             117 non-null    object 

# Converted the Datatype of `Equity` column

In [73]:
df

,Episode Number,Pitch Number,Brand,Idea,Investment Amount (In Lakhs INR),Debt In lakhs INR,Equity,Anupam,Ashneer,Namita,Aman,Peyush,Vineeta,Ghazal,Season,Sharks Invested
49,17,50,Find Your Kicks India,Sneaker Resale,50,0,25.0,Y,Y,Y,Y,Y,N,N,1,5
63,20,64,IN A CAN,Can Cocktails,10,0,10.0,Y,Y,Y,Y,Y,N,N,1,5
15,6,16,Skippi Pops,Ice-Pops,10,0,15.0,Y,Y,Y,Y,N,Y,N,1,5
79,25,80,Sunfox Technologies,Portable ECG Device,10,0,6.0,Y,N,Y,N,Y,Y,Y,1,5
58,19,59,WeSTOCK,Livestock health monitoring AI,60,0,10.0,N,Y,Y,Y,Y,N,N,1,4
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
40,14,41,Morikko Pure Foods,Healthy Food Snacks,0,0,0.0,N,N,N,N,N,N,N,1,0
72,23,73,Urban Monkey,Streetwear,0,0,0.0,N,N,N,N,N,N,N,1,0
73,23,74,Guardian Gears,Motorcycle Luggage,0,0,0.0,N,N,N,N,N,N,N,1,0
74,23,75,Modern Myth,Bags,0,0,0.0,N,N,N,N,N,N,N,1,0


In [100]:
top_deals = df.sort_values(by='Investment Amount (In Lakhs INR)', ascending=False).head(5)
print(top_deals[['Brand','Investment Amount (In Lakhs INR)']])

                   Brand  Investment Amount (In Lakhs INR)
95          Kabaddi Adda                                80
52            RoadBounce                                80
0    BluePine Industries                                75
33        Auli Lifestyle                                75
100             AyuRythm                                75


# Top 5 Largest Deals

# For analyzing founder profiles, we don’t have a separate founder column, but we can analyze founder credibilit by creating a `Shark Invested` column.

In [34]:
shark_cols = ['Anupam','Ashneer','Namita','Aman','Peyush','Vineeta','Ghazal']
df['Sharks Invested'] = (df[shark_cols] == 'Y').sum(axis=1)

In [43]:
df.sort_values(by='Sharks Invested', ascending = False).head(4)

,Episode Number,Pitch Number,Brand,Idea,Investment Amount (In Lakhs INR),Debt (In lakhs INR),Equity,Anupam,Ashneer,Namita,Aman,Peyush,Vineeta,Ghazal,Season,Sharks Invested
49,17,50,Find Your Kicks India,Sneaker Resale,50,0,25.0,Y,Y,Y,Y,Y,N,N,1,5
15,6,16,Skippi Pops,Ice-Pops,10,0,15.0,Y,Y,Y,Y,N,Y,N,1,5
79,25,80,Sunfox Technologies,Portable ECG Device,10,0,6.0,Y,N,Y,N,Y,Y,Y,1,5
63,20,64,IN A CAN,Can Cocktails,10,0,10.0,Y,Y,Y,Y,Y,N,N,1,5


# Startups with more sharks investing shows more convincing founders or ideas.

In [42]:
df.groupby('Sharks Invested')['Equity'].mean()

Sharks Invested
0     0.000000
1    18.746818
2    16.115000
3    14.500000
4    14.800000
5    14.000000
Name: Equity, dtype: float64

# It shows the average equity given for startups with 1, 2, 3, 4, and 5 sharks investing.

In [90]:
df.groupby('Episode Number')['Investment Amount (In Lakhs INR)'].mean()

Episode Number
1     46.666667
2     23.333333
3     31.666667
4     41.666667
5      3.333333
6     20.000000
7      3.333333
8     45.333333
9     33.333333
10    28.333333
11    26.666667
12    38.333333
13    23.333333
14     3.333333
15    13.333333
16    42.500000
17    36.250000
18     0.250000
19    15.000000
20    16.666667
21    23.333333
22    15.000000
23    12.500000
24    20.000000
25    20.000000
26    38.333333
27     5.000000
28    20.000000
29    25.000000
30    20.000000
31    18.750000
32    12.500000
33    27.500000
34    10.000000
35    13.333333
Name: Investment Amount (In Lakhs INR), dtype: float64

# Shows the amount invested in different episodes over time

# For analyzing funding stage success, we don’t have a separate funding stage column, but we can analyze simple proxy using Investment Amount for funding stage.

In [92]:
def funding_stage(amount):
    if amount == 0:
        return 'No Deal'
    elif amount < 10:
        return 'Seed'
    elif amount < 50:
        return 'Early'
    elif amount < 100:
        return 'Growth'
    else:
        return 'Expansion'

df['Funding Stage'] = df['Investment Amount (In Lakhs INR)'].apply(funding_stage)

In [95]:
df

,Episode Number,Pitch Number,Brand,Idea,Investment Amount (In Lakhs INR),Debt (In lakhs INR),Equity,Anupam,Ashneer,Namita,Aman,Peyush,Vineeta,Ghazal,Season,Funding Stage
0,1,1,BluePine Industries,Frozen Momos,75,0,18%,N,Y,N,Y,N,Y,N,1,Growth
1,1,2,Booz scooters,Renting e-bike for mobility in private spaces,40,0,50%,N,Y,N,N,N,Y,N,1,Early
2,1,3,Heart up my Sleeves,Detachable Sleeves,25,0,30%,Y,N,N,N,N,Y,N,1,Early
3,2,4,Tagz Foods,Healthy Potato Chips,70,0,2.75%,N,Y,N,N,N,N,N,1,Growth
4,2,5,Head and Heart,Brain Development Course,0,0,0,N,N,N,N,N,N,N,1,No Deal
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
112,34,113,Green Protein,Plant-Based Protein,0,0,0,N,N,N,N,N,N,N,1,No Deal
113,34,114,On2Cook,Fastest Cooking Device,0,0,0,N,N,N,N,N,N,N,1,No Deal
114,35,115,Jain Shikanji,Lemonade,40,0,30.00%,Y,Y,N,Y,N,Y,N,1,Early
115,35,116,Woloo,Washroom Finder,0,0,0,N,N,N,N,N,N,N,1,No Deal


In [97]:
df['Funding Stage'].value_counts()

Funding Stage
No Deal    52
Early      33
Growth     30
Seed        2
Name: count, dtype: int64

# Most startups on Shark Tank India did not receive any investment.
# Other than "No Deal", The majority were in the “Early” and “Growth” stages.
# Where as a very few startups received “Seed” stage fundings.

In [104]:
df.to_csv("C:/Users/user/Desktop/Elevate Labs Internship/Projects/Project_1/sharktank_cleaned.csv", index=False)